crates/walrus-sdk/src/client/quilt_client.rs

```diff
@@ -0,0 +1,583 @@
+                );
+                refs.extend(all_slivers.iter());
+                decoder.add_slivers(&refs);
+                decoder.get_or_decode_quilt_index()?
+            }
+            Err(e) => return Err(e.into()),
+        };
+
+        Ok(quilt_index)
+    }
+
+    /// Retrieves the quilt patches of the given identifiers from the quilt.
+    pub async fn get_blobs_by_identifiers(
+        &self,
+        quilt_id: &BlobId,
+        identifiers: &[&str],
+    ) -> ClientResult<Vec<QuiltStoreBlobOwned>> {
+        let metadata = self.get_quilt_metadata(quilt_id).await?;
+
+        let blobs = match metadata {
+            QuiltMetadata::V1(metadata) => {
+                self.get_blobs_by_identifiers_impl::<QuiltVersionV1>(
+                    &metadata.get_verified_metadata(),
+                    &metadata.index.into(),
+                    identifiers,
+                )
+                .await?
+            }
+        };
+
+        Ok(blobs)
+    }
+
+    /// Retrieves quilt patches from QuiltV1 by identifiers.
+    async fn get_blobs_by_identifiers_impl<V: QuiltVersion>(
+        &self,
+        metadata: &VerifiedBlobMetadataWithId,
+        index: &QuiltIndex,
+        identifiers: &[&str],
+    ) -> ClientResult<Vec<QuiltStoreBlobOwned>>
+    where
+        SliverData<V::SliverAxis>: TryFrom<Sliver>,
+    {
+        // Retrieve slivers for the given identifiers.
+        let sliver_indices = index.get_sliver_indices_for_identifiers(identifiers)?;
+        let (certified_epoch, _) = self
+            .client
+            .get_blob_status_and_certified_epoch(metadata.blob_id(), None)
+            .await?;
+        let retrieved_slivers = self
+            .client
+            .retrieve_slivers_with_retry::<V::SliverAxis>(
+                metadata,
+                &sliver_indices,
+                certified_epoch,
+                Self::MAX_RETRIEVE_SLIVERS_ATTEMPTS,
+                Self::TIMEOUT_DURATION,
+            )
+            .await;
+
+        if let Ok(slivers) = retrieved_slivers {
+            let sliver_refs: Vec<_> = slivers.iter().collect();
+            let decoder = V::QuiltConfig::get_decoder_with_quilt_index(&sliver_refs, index);
+            identifiers
+                .iter()
+                .map(|identifier| {
+                    decoder
+                        .get_blob_by_identifier(identifier)
+                        .map_err(ClientError::other)
+                })
+                .collect::<Result<Vec<_>, _>>()
+        } else {
+            let quilt = self.get_quilt_enum(metadata, certified_epoch).await?;
+            identifiers
+                .iter()
+                .map(|identifier| {
+                    quilt
+                        .get_blob_by_identifier(identifier)
+                        .map_err(ClientError::other)
+                })
+                .collect::<Result<Vec<_>, _>>()
+        }
+    }
+
+    /// Retrieves the quilt from Walrus.
+    async fn get_quilt_enum(
```

Comment:
Nit: The function name is a bit confusing. How about something like `get_full_quilt`?